mod/groups/views/default/groups/profile/fields.php

```diff
@@ -3,32 +3,43 @@
  * Group profile fields
  */
 
-$group = $vars['entity'];
+$group = elgg_extract('entity', $vars);
+if (!$group instanceof ElggGroup) {
+	return;
+}
 
 $profile_fields = elgg_get_config('group');
+if (empty($profile_fields) || !is_array($profile_fields)) {
+	return;
+}
 
-if (is_array($profile_fields) && count($profile_fields) > 0) {
-	foreach ($profile_fields as $key => $valtype) {
-		// do not show the name
-		if ($key == 'name') {
-			continue;
-		}
-
-		$value = $group->$key;
-		if (is_null($value)) {
-			continue;
-		}
+foreach ($profile_fields as $key => $valtype) {
+	// do not show the name
+	if ($key == 'name') {
+		continue;
+	}
 
-		$options = ['value' => $group->$key];
-		if ($valtype == 'tags') {
-			$options['tag_names'] = $key;
-		}
+	$value = $group->$key;
+	if (is_null($value)) {
```

Comment:
do we also want to skip `''`?

Reply:
maybe... but leaving it as is for now... empty strings are not saved when editing the profile